perf(rolldown): avoid allocate the sourcemap sources on non Windows os

Brooooooklyn opened the pull request:
<img width="1858" height="1916" alt="CleanShot 2025-10-17 at 14 24 19@2x" src="https://github.com/user-attachments/assets/19df558c-c928-4a13-bb6b-30e61e008a53" />

Brooooooklyn commented:
* **#6590** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6590?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6590?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#6589** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6589?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite: merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/rolldown?ref=merge-queue-instructions-in-stack-comment&prId=6590).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

hyf0 commented:
### Merge activity

* **Oct 17, 3:02 AM UTC**: The merge label 'graphite: merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) once it meets the requirements.
* **Oct 17, 3:16 AM UTC**: [Graphite](https://app.graphite.dev/github/pr/rolldown/rolldown/6590) rebased this pull request, because this pull request is set to merge when ready.
* **Oct 17, 6:11 AM UTC**: `hyf0` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).
* **Oct 17, 6:25 AM UTC**: Merged by the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).

hyf0 commented on crates/rolldown/src/utils/process_code_and_sourcemap.rs:
```diff
@@ -57,11 +58,25 @@ pub async fn process_code_and_sourcemap(
-    }
+    let sources = try_join_all(map.get_sources().map(async |source| {
+      let source =
+        sourcemap_path_transform.call(source, map_path.to_string_lossy().as_ref()).await?;
```
Does this one actually make sense? @Brooooooklyn cc

Brooooooklyn replied:
Yes, edited

Brooooooklyn committed:
perf(rolldown): avoid allocate the sourcemap sources on non Windows os (#6590)

<img width="1642" height="1844" alt="CleanShot 2025-10-17 at 14 07 15@2x" src="https://github.com/user-attachments/assets/594a3225-44d4-41dc-a344-1ef567a80ba7" />
